docs/custom-filters.adoc

```diff
@@ -45,23 +149,163 @@ Instances are specific to the channel between a single client and a single broke
 
 It exists while the client remains connected.
 
+=== Filter Patterns
+
+Kroxylicious Protocol Filters support several patterns:
+
+1. <<Synchronously intercepting/mutating Requests and Responses>>
```

Comment:
Though I'm not sure `synchronously` is quite the right term, in my mental model its more a case of in-line (or maybe in band) observation or transformation.

Reply:
Have removed that language from the title, agree it's not really a good description of the pattern. Though I still have used the term once when describing the contract like:

```Note you should not attempt to call `forward` methods on the `KrpcFilterContext` using the CompletionStage, they still
must be called synchronously during the invocation of the `onXyz[Request|Response]` method.```

Reply:
I really think we need an asynchronous API (#351) but I agree documenting how things stand now makes sense.